Altis_Life.Altis/core/functions/fn_fetchVehInfo.sqf

```diff
@@ -21,37 +21,28 @@
     11: Max Horse power
     12: Fuel Capacity
 */
-private ["_class","_scope","_picture","_displayName","_vehicleClass","_side","_faction","_superClass","_speed","_armor","_seats","_hp","_fuel"];
-_class = [_this,0,"",[""]] call BIS_fnc_param;
+if !(params [
+    ["_class","",[""]]
+]) exitWith {[]}; //Bad params
 if (_class isEqualTo "") exitWith {[]}; //Bad class passed.
 if (!isClass (configFile >> "CfgVehicles" >> _class)) exitWith {[]}; //Class doesn't exist in CfgVehicles
 
-//Predefine some stuff.
-_scope = -1;
-_picture = "";
-_displayName = "";
-_vehicleClass = "";
-_side = -1;
-_faction = "";
-_speed = 0;
-_armor = 0;
-_seats = 0;
-_hp = 0;
-_fuel = 0;
+//prepare config
+private _config = (configFile >> "CfgVehicles" >> _class);
 
 //Fetch
-_scope = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"scope");
-_picture = FETCH_CONFIG2(getText,"CfgVehicles",_class,"picture");
-_displayName = FETCH_CONFIG2(getText,"CfgVehicles",_class,"displayName");
-_vehicleClass = FETCH_CONFIG2(getText,"CfgVehicles",_class,"vehicleClass");
-_side = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"side");
-_faction = FETCH_CONFIG2(getText,"CfgVehicles",_class,"faction");
-_superClass = BASE_CONFIG("CfgVehicles",_class);
-_speed = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"maxSpeed");
-_armor = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"armor");
-_seats = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"transportSoldier");
-_hp = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"enginePower");
-_fuel = FETCH_CONFIG2(getNumber,"CfgVehicles",_class,"fuelCapacity");
+private _scope = getNumber(_config >> "scope");
+private _picture = getText(_config >> "picture");
+private _displayName = getText(_config >> "picture");
+private _vehicleClass = getText(_config >> "vehicleClass");
+private _side = getNumber(_config >> "side");
+private _faction = getText(_config >> "faction");
+private _superClass = inheritsFrom _config;
+private _speed = getNumber(_config >> "maxSpeed");
+private _armor = getNumber(_config >> "armor");
+private _seats = getNumber(_config >> "transportSoldier") + count ("true" configClasses (_config >> "Turrets")); //number of seats = number of passengers + number of vehicle turrets
```

Comment:
Before it didn't count the driver seat, gunner seat, commander seat, and any other turret seat.
Now it counts every seat, including drone AI, minus one for the driver.

Both seem strange, so I am not sure what this is supposed to be.
The best way to count all seats usually is to use `count fullCrew [_vehicle, "", true]`, but that doesn't work on classnames.

Reply:
I get that one with AI, but the Problem with not counting turrets is, that with most vehicles the number of seats is otherwise totally incorrect. 
Example: MB4 WD - old function returns 0, actual number of seats is 3 + 1 driver. new function returns 3
Prowler LMG - old function returns 0, actual number of seats is 4 + 1 driver, new functions returns 4

Most of the vanilla vehicles now have "Firing from vehicle" implemented, which means they changed "normal" seats to turrets to accomplish this. That's why I changed it to count turrets too.